src/ui/speechrecognition/voicesearchcontroller.js

```diff
@@ -66,6 +69,7 @@ export default class VoiceSearchController {
       });
     } else if (this._state === State.LISTENING) {
       this._enterNotListeningState();
+      return Promise.resolve();
```

Comment:
This technically "could" return undefined instead of a promise if this._state somehow isn't either of the 2 states
could we move return Promise.resolve() outside the statements? We could also use async await though I kind of like traditional promises

Reply:
true. updated!